ircd/newconf.c

```diff
@@ -1542,6 +1542,21 @@ conf_set_general_stats_i_oper_only(void *data)
 		conf_report_error("Invalid setting '%s' for general::stats_i_oper_only.", val);
 }
 
+static void
+conf_set_general_stats_l_oper_only(void *data)
+{
+	char *val = data;
+
+	if(rb_strcasecmp(val, "yes") == 0)
+		ConfigFileEntry.stats_l_oper_only = 2;
+	else if(rb_strcasecmp(val, "self") == 0)
+		ConfigFileEntry.stats_l_oper_only = 1;
+	else if(rb_strcasecmp(val, "no") == 0)
+		ConfigFileEntry.stats_l_oper_only = 0;
+	else
+		conf_report_error("Invalid setting '%s' for general::stats_l_oper_only.", val);
```

Comment:
I'd prefer if the error messages also listed what the valid values are when the set of them is small, but I looked around and the rest of the configuration processing behaves this same way, so that's something we could worry about separately.